session.go

```diff
@@ -49,23 +51,32 @@ func NewSession(url string, username string, password string) (session *Session,
 		return
 	}
 
+	err = session.login(username, password)
+	if err != nil {
+		return
+	}
+
+	return
+}
+
+func (c *Session) login(username, password string) error {
 	// login to API
 	params := map[string]string{
 		"user":     username,
 		"password": password,
 	}
 
-	res, err = session.Do(NewRequest("user.login", params))
+	res, err := c.Do(NewRequest("user.login", params))
 	if err != nil {
-		return nil, fmt.Errorf("Error logging in to Zabbix API: %v", err)
```

Comment:
I think the addition context here is important to keep. Otherwise, the caller might not know what request caused the API error (i.e. login).